utils/python/CIME/XML/generic_xml.py

```diff
@@ -63,55 +58,72 @@ def write(self, outfile=None):
             outfile = self.filename
 
         logger.debug("write: " + outfile)
-        xmlstr = ET.tostring(self.root)
-        doc = minidom.parseString(xmlstr)
-        with open(outfile,'w') as xmlout:
-            doc.writexml(xmlout,addindent='  ')
+        try:
+            xmlstr = ET.tostring(self.root)
+        except:
+            ET.dump(self.root)
+            expect(False, "Could not write file %s, xml formatting error"%self.filename)
+
+        # xmllint provides a better format option for the output file
+        xmllint = which("xmllint")
```

Comment:
If xmllint exists in the path use it to format the output xml.   This makes the file indentation nicer.